drivers/usb/device/usb_dc_it82xx2.c

```diff
@@ -0,0 +1,1687 @@
+#define SC_ACK_RXED			BIT(6)
+#define SC_DATA_SEQUENCE		BIT(7)
+
+/* SC_INTERRUPT_STATUS_REG */
+#define SC_TRANS_DONE			BIT(0)
+#define SC_RESUME_INT			BIT(1)
+#define SC_RESET_EVENT			BIT(2)
+#define SC_SOF_RECEIVED			BIT(3)
+#define SC_NAK_SENT_INT			BIT(4)
+#define VBUS_DET_INT			BIT(5)
+
+/* SC_CONTROL_REG */
+#define SC_GLOBAL_ENABLE		BIT(0)
+#define SC_TX_LINE_STATE_DM		BIT(1)
+#define SC_TX_LINE_STATE_DP		BIT(2)
+#define SC_DIRECT_CONTROL		BIT(3)
+#define SC_FULL_SPEED_LINE_POLARITY	BIT(4)
+#define SC_FULL_SPEED_LINE_RATE		BIT(5)
+#define SC_CONNECT_TO_HOST		BIT(6) /* internal pull-up */
+
+/* ENDPOINT[3..0]_CONTROL_REG */
+#define ENDPOINT_EN			BIT(0)
+#define ENDPOINT_RDY			BIT(1)
+#define EP_OUTDATA_SEQ			BIT(2)
+#define EP_SEND_STALL			BIT(3)
+#define EP_ISO_ENABLE			BIT(4)
+#define EP_DIRECTION			BIT(5)
+
+/**
+ *
+ * Some definitions of related structures and states of the USB device
+ *
+ */
+enum usb_device_state {
+	/* NOTATTACHED isn't in the USB spec, and this state acts
+	 * the same as ATTACHED ... but it's clearer this way.
+	 */
+	USB_STATE_NOTATTACHED = 0,
+
+	/* chapter 9 and authentication (wireless) device states */
+	USB_STATE_ATTACHED,
+	USB_STATE_POWERED,         /* wired */
+	USB_STATE_RECONNECTING,    /* auth */
+	USB_STATE_UNAUTHENTICATED, /* auth */
+	USB_STATE_DEFAULT,         /* limited function */
+	USB_STATE_ADDRESS,
+	USB_STATE_CONFIGURED,      /* most functions */
+
+	USB_STATE_SUSPENDED
+
+	/* NOTE:  there are actually four different SUSPENDED
+	 * states, returning to POWERED, DEFAULT, ADDRESS, or
+	 * CONFIGURED respectively when SOF tokens flow again.
+	 * At this level there's no difference between L1 and L2
+	 * suspend states.  (L2 being original USB 1.1 suspend.)
+	 */
+};
+
+enum ep_status {
+	EP_INIT,
+	EP_CHECK,
+	EP_CONFIG,
+	EP_CONFIG_IN,
+	EP_CONFIG_OUT,
+};
+
+enum trans_type {
+	NONE_TOKEN,
+	SETUP_TOKEN,
+	IN_TOKEN,
+	OUT_TOKEN,
+};
+
+enum setup_stage {
+	INIT_ST,
+	SETUP_ST,
+	DIN_ST, /* Data In Stage */
+	DOUT_ST, /* Data Out Stage */
+	STATUS_ST,
+	STALL_SEND,
+};
+
+enum extend_ep_ctrl {
+	/* EPN0N1_EXTEND_CONTROL_REG */
+	EXT_EP_ISO_DISABLE,
+	EXT_EP_ISO_ENABLE,
+	EXT_EP_SEND_STALL,
+	EXT_EP_CLEAR_STALL,
+	EXT_EP_CHECK_STALL,
+	EXT_EP_DATA_SEQ_1,
+	EXT_EP_DATA_SEQ_0,
+	EXT_EP_DATA_SEQ_INV,
+	/* EPN_EXTEND_CONTROL1_REG */
+	EXT_EP_DIR_IN,
+	EXT_EP_DIR_OUT,
+	EXT_EP_ENABLE,
+	EXT_EP_DISABLE,
+	/* EPN_EXTEND_CONTROL2_REG */
+	EXT_EP_READY,
+};
+
+struct usb_it8xxx2_wuc {
+	/* WUC control device structure */
+	const struct device *wucs;
+	/* WUC pin mask */
+	uint8_t mask;
+};
+
+struct usb_it82xx2_config {
+	const struct pinctrl_dev_config *pcfg;
+	const struct usb_it8xxx2_wuc *wuc_list;
+};
+
+static const struct usb_it8xxx2_wuc usb_wuc0[IT8XXX2_DT_INST_WUCCTRL_LEN(0)] =
+		IT8XXX2_DT_WUC_ITEMS_LIST(0);
+
+PINCTRL_DT_INST_DEFINE(0);
+
+static const struct usb_it82xx2_config ucfg0 = {
+	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(0),
+	.wuc_list = usb_wuc0
+};
+
+struct usb_device_ep_data {
```

Comment:
`struct it82xx2_endpoint_data {`

Reply:
Got it, sir. I've renamed it with prefix.